### PR TITLE
chore(pipeline) fix deployment of the Linux Docker images to properly support multiple architectures

### PR DESCRIPTION
**Pull request by smerle33**
This PR ensures that a Docker multi-architecture manifest is created and published, to provide both `amd64` and `arm64` for each release.

### CODE REVIEW

File: Jenkinsfile_k8s [hunk 1]
```diff
@@ -242,12 +242,14 @@ pipeline {
                   // Execute build only for this matrix cell's setup
                   retry(count: 2, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
                     sh 'packer build -timestamp-ui -force -only="${PKR_VAR_image_type}.${PKR_VAR_agent_os_type}" ./'
+                    // adding manually a cpu architecture tag to the docker image
+                    sh 'docker tag jenkinsciinfra/jenkins-agent-${PKR_VAR_agent_os_type}:${PKR_VAR_architecture}'
```
Comment (dduportal) [inline Jenkinsfile_k8s:246]:
```suggestion
                    sh 'docker tag "jenkinsciinfra/jenkins-agent-${agent_type}:latest" "jenkinsciinfra/jenkins-agent-${PKR_VAR_agent_os_type}:${PKR_VAR_architecture}"'
```

- `docker tag` command requires 2 arguments: source image (e.g. an image already present in the output of `docker image ls` so it's the image created by `packer`) and new image name (with tag) to add as alias.

Comment (smerle33) [reply Jenkinsfile_k8s:246]: at this point of the pipeline, can I use agent_type or should I remerge with the 2 variables ? : 
```
final String pkr_var_agent_os_type = agent_type.split('-')[0]
final String pkr_var_agent_os_version = agent_type.split('-')[1]
```

**Comment by smerle33** [comment]
https://infra.ci.jenkins.io/blue/organizations/jenkins/infra-tools%2Fpacker-images/detail/PR-509/15/pipeline/232
only packer crash that can be discarded ...

File: Jenkinsfile_k8s [hunk 2]
```diff
@@ -260,5 +265,32 @@ pipeline {
...
       }
     }
+    stage('Build Docker Manifest') {
+      when {
+        expression {
+          return env.TAG_NAME != null
+        }
+      }
+      environment {
+        agent_type = 'ubuntu-20.04'
```
Comment (dduportal) [inline Jenkinsfile_k8s:275]:
```suggestion
        // Static variable definition as this stage is outside the matrix scope
        // Improvement: pass dynamically the list of images from the matrix (e.g. use full scripted pipeline) to support other Docker agent types (such as windows-2019 or windows-2022)
        agent_type = 'ubuntu-20.04'
```

File: Jenkinsfile_k8s [hunk 3]
```diff
@@ -260,5 +265,32 @@ pipeline {
...
+      }
+      steps {
+        script {
+          infra.withDockerPushCredentials {
+            sh 'docker manifest create \
+                jenkinsciinfra/jenkins-agent-${agent_type}:latest \
+                --amend jenkinsciinfra/jenkins-agent-${agent_type}:arm64 \
+                --amend jenkinsciinfra/jenkins-agent-${agent_type}:amd64'
+            sh 'docker manifest push jenkinsciinfra/jenkins-agent-${agent_type}:latest'
```
Comment (dduportal) [inline Jenkinsfile_k8s:284]:
```suggestion
            sh 'docker manifest push jenkinsciinfra/jenkins-agent-"${agent_type}":latest'
```

File: Jenkinsfile_k8s [hunk 4]
```diff
@@ -260,5 +265,32 @@ pipeline {
...
+
+            sh 'docker manifest create \
+                jenkinsciinfra/jenkins-agent-${agent_type}:${TAG_NAME} \
+                --amend jenkinsciinfra/jenkins-agent-${agent_type}:arm64 \
+                --amend jenkinsciinfra/jenkins-agent-${agent_type}:amd64'
+            sh 'docker manifest push jenkinsciinfra/jenkins-agent-${agent_type}:${TAG_NAME}'
```
Comment (dduportal) [inline Jenkinsfile_k8s:290]:
```suggestion
            sh 'docker manifest push jenkinsciinfra/jenkins-agent-"${agent_type}":"${TAG_NAME}"'
```
- nitpick for the quotes around the "agent_type" variable
- Not nitpick, but not blocker either: it's safer to "protect" variables such as `TAG_NAME` that could contains random characters (depends on the value the user set for the tag)

**Pull request merged**